apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -53,13 +53,22 @@ export async function updatePriority(
 		}),
 	};
 
-	const created = await LivechatPriority.findOneAndUpdate(query, update, {
-		returnDocument: 'after',
-	});
+	try {
+		const created = await LivechatPriority.findOneAndUpdate(query, update, {
+			returnDocument: 'after',
+		});
 
-	if (!created.ok || !created.value) {
-		throw Error('Error updating priority');
-	}
+		if (!created.ok || !created.value) {
+			throw Error('Error updating priority');
+		}
 
-	return created.value;
+		return created.value;
+	} catch (error: any) {
+		// check if its a duplicate key error
+		if (error?.code === 11000) {
+			throw new Error('Error! Priority already exists with this name');
+		}
+
+		throw error;
+	}
```

Comment:
The index works, but, what if instead of trying to guess the error, we're more explicit and do a `find` b4 to check? That way, we can also control casing of the word as harmeet reported 👀 

(We can leave the index for safety tho)